### DISCUSSION TITLE
What information is only available from Ergast

### DISCUSSION THREAD

**Discussion by Casper-Guo** [theOehrly/Fast-F1#618] [Q&A] [answered]
For sessions where F1API is available (2018 onwards), what are we sourcing from Ergast and not available otherwise? Trying to figure out what I need to handle when Ergast is down.

Specifically, sometimes session results seem to contain incorrect finishing positions. Was wondering if that's ergast related.

**Comment by theOehrly** [comment, marked as answer]
The logic behind that has a lot of conditions but mostly:

- 'Abbreviation', 'FirstName', 'LastName', 'TeamName', 'FullName', 'DriverNumber', 'BroadcastName', 'TeamColor', 'CountryCode', 'HeadshotUrl' are taken from the livetiming api
- Ergast-specific 'DriverId' and 'TeamId', 'DriverNumber' (as reference for merging data), 'Position', 'ClassifiedPosition', 'GridPosition', 'Status', 'Points', 'Time', 'Q1', 'Q2', 'Q3' are taken from Ergast

- Special case, if a driver did not start (DNS), 'Abbreviation', 'FirstName', 'LastName', 'TeamName', 'FullName', 'DriverNumber' are taken from Ergast as well, because the F1 livetiming API does not list those drivers

- There is a fallback where 'Position', 'Q1', 'Q2', 'Q3' are calculated from timing data. In case of Qualifying this only happens if Ergast is down. For Sprint Qualifying, this always happens because Ergast does not support SQ.

- Only if the session is a race: the lap times of the first lap as they are unavailable through the livetiming API

I think that should be all.

**Reply by Casper-Guo** [reply]
For my specific case then, are session results taken from F1 API directly or from Ergast with fallback?

**Reply by theOehrly** [reply]
If by results you mean Position, Points, .... this is always taken from Ergast or calculated with the fallback. The F1 API does not provide this information.

Incorrect finishing positions are therefore either incorrect on Ergast or a bug in the fallback calculation.

**Reply by Casper-Guo** [reply]
I'll keep an eye out for the next time I notice any abnormality. It's very rare but might reveal some bugs in the fallback calculation like you said
